src/html_highlighter.js

```diff
@@ -1087,14 +1087,17 @@
    * @param {TextContent} content - Reference to <code>TextContent</code>
    * instance.
    * @param {string} subject - Subject string to match. */
+  /* FIXME: create a class for matching of regular expression subjects. */
   var TextFinder = function(content, subject)
   {
     /* Construct base class. */
     Finder.call(this, content);
 
     /* Build an array containing all hits of `subject´. */
     var match,
-        re = new RegExp(subject.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&"),
+        re = subject instanceof RegExp
+           ? subject
+           : new RegExp(subject.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&"),
```

Comment:
A regex to escape a regex!? Hah. It is indeed [recommended by MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions?redirectlocale=en-US&redirectslug=JavaScript%2FGuide%2FRegular_Expressions), wow.